modules/programs/neovim.nix

```
@@ -392,26 +392,25 @@ in {
 
     home.sessionVariables = mkIf cfg.defaultEditor { EDITOR = "nvim"; };
 
-    xdg.configFile =
-      let hasLuaConfig = hasAttr "lua" config.programs.neovim.generatedConfigs;
-      in mkMerge (
-        # writes runtime
-        (map (x: x.runtime) pluginsNormalized) ++ [{
-          "nvim/init.lua" = let
-            luaRcContent =
-              lib.optionalString (neovimConfig.neovimRcContent != "")
-              "vim.cmd [[source ${
-                pkgs.writeText "nvim-init-home-manager.vim"
-                neovimConfig.neovimRcContent
-              }]]" + config.programs.neovim.extraLuaConfig
-              + lib.optionalString hasLuaConfig
-              config.programs.neovim.generatedConfigs.lua;
-          in mkIf (luaRcContent != "") { text = luaRcContent; };
-
-          "nvim/coc-settings.json" = mkIf cfg.coc.enable {
-            source = jsonFormat.generate "coc-settings.json" cfg.coc.settings;
-          };
-        }]);
+    xdg.configFile = let
+      hasLuaConfig = hasAttr "lua" config.programs.neovim.generatedConfigs;
+      vimRcContent = lib.optionalString (neovimConfig != "") pkgs.writeText
```

It seems like this is what it used to be...

```suggestion
      vimRcContent = lib.optionalString (neovimConfig.neovimRcContent != "") pkgs.writeText
```